framework/src/tinyalsa/tinyalsa.c

```diff
@@ -1292,22 +1365,41 @@ int pcm_wait(struct pcm *pcm, int timeout)
 	unsigned int size;
 	int prio;
 	struct timespec st_time;
+	int count = 0;
 
 	if (pcm == NULL) {
 		return -EINVAL;
 	}
-	if ((pcm->flags & PCM_OUT) && pcm->buf_ptr < pcm->buffer_cnt) {
-		/* In playback scenario, if audio buffers are available, return immediately */
-		pcm->next_buf = pcm->pBuffers[pcm->buf_ptr];
-		pcm->next_size = pcm->pBuffers[pcm->buf_ptr]->nmaxbytes;
-		return 1;
-	} else {
-		/* We dont have any empty buffers. wait for deque message from kernel */
+
+	/* If there are some messages in the queue, empty them */
+	while (1) {
+		clock_gettime(CLOCK_REALTIME, &st_time);
+		size = mq_timedreceive(pcm->mq, (FAR char *)&msg, sizeof(msg), &prio, &st_time);
+		if (size != sizeof(msg)) {
+			break;
+		}
+
+		/* If you find a dequeue message in the queue,
+		update the apb flags and curbyte and increment the count*/
+		if (msg.msgId == AUDIO_MSG_DEQUEUE) {
+			apb = (struct ap_buffer_s *)msg.u.pPtr;
+			apb->flags &= ~AUDIO_APB_OUTPUT_ENQUEUED;
+			apb->curbyte = 0;
+			count++;
+		}
+	}
+
+	/* If there were no buffers in the queue, wait for codec to put a buffer on the queue */
+	if (count == 0) {
```

Comment:
How about below code?
/* Buffer is in the queue, hence we don't have to wait */
if (count > 0) {
    return 1;
}

if (timeout > 0){
............
} else {
............
}

Reply:
Sure. I changed it.